### PR TITLE
✨ gains `sum_with_threads`

### LINKED ISSUE #3: 👷 add CI infrastructure to benchmark a `blazr::` function against it's `tidyverse::` equivalent
A key goal of the `blazr` package is to prioritize performance gains (memory, speed) against `tidyverse` counterparts. 
It would be helpful to build supporting CI infrastructure to run (and report) this benchmarking for us.

### LINKED ISSUE #15: ✨ gains exported function `sum_with_threads`
Acceptance criteria:
- [x] `sum_with_threads(x, n)
- [x] `x` - A vector of integers
- [x] `n` - An integer, indicating the number of threads 
- [x] `@return` - An integer, the sum of each element in the input vectors

The goal of implementing this is to work out how multi-threading _might_ work using `savvy`, with a very simple canonical example. 

The underlying Rust process would split the sum process into `n` threads, and compute the piece-wise sum of each, before summing each piece and returning the result.

### PR DESCRIPTION
**Pull request by jdhoffa**
In this PR, I add (and export) a very simple multi-threaded implementation of the `sum` function. 
Currently, the sum only works on integers. A cool TODO for someone could be to extend the functionality to any generic numeric. 

As per the discussion in #15, I started by writing the pure Rust functionality (see `sum_with_threads_impl` in `src/rust/src/lib.rs`). I followed test-driven and compiler-driven development, and worked to get some simple unit tests demonstrating the functionality to pass. 

Once those passed, I wrote an interface wrapper to give `R` and API to the `Rust` implementation function. 

Once that was done, I ran `savvy::savvy_update()` and `devtools::document()`, and then demo'd the new functionality in the `README.Rmd`.

## Note to reviewers
The git log of this PR gives a step-by-step of the development process I followed here. Might be useful for contributors wanting to add new functionality. 

⚠️ I also removed all of the boilerplate exported functions that were auto-populated by `savvy`

Closes #15 
Maybe closes #3 (sort of... I think it was ambitious to benchmark against `dplyr`, but this adds benchmarking directly to the `README.md`)

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -2,7 +2,5 @@
 
 S3method("$<-",savvy_blazr__sealed)
 S3method("[[<-",savvy_blazr__sealed)
-S3method(print,Person__bundle)
-export(int_times_int)
-export(to_upper)
```
Comment (jdhoffa) [inline NAMESPACE:7]: `int_times_int`, `to_upper` and the `Person` S3 object are all default implementations from `savvy`. I stripped out that boilerplate code.

File: src/rust/api.h [hunk 2]
```diff
@@ -1,8 +1 @@
-SEXP savvy_to_upper__ffi(SEXP c_arg__x);
-SEXP savvy_int_times_int__ffi(SEXP c_arg__x, SEXP c_arg__y);
-
-// methods and associated functions for Person
-SEXP savvy_Person_new__ffi(void);
-SEXP savvy_Person_set_name__ffi(SEXP self__, SEXP c_arg__name);
-SEXP savvy_Person_name__ffi(SEXP self__);
-SEXP savvy_Person_associated_function__ffi(void);
\ No newline at end of file
+SEXP savvy_sum_with_threads__ffi(SEXP c_arg__x, SEXP c_arg__n);
```
Comment (jibarozzo) [inline src/rust/api.h:1]: What was the difference between `*__impl` and `*__ffi` ?

**Review by jibarozzo — APPROVED** [review]
This looks great. I'll work on it to include all numerical and see how it goes.

File: src/rust/api.h (hunk 2 above)
Comment (jdhoffa) [reply src/rust/api.h:1]: I have no idea 😅 

Everything that was added in https://github.com/r-staceans/blazr/pull/19/commits/1df48549f1998c732c889c86cc75174209b105b8 was included automatically when running `savvy::savvy_update()`, so I'm not 100% sure how it all works under the hood.

Comment (jdhoffa) [reply src/rust/api.h:1]: More precisely: 
The `*_impl` function was my design/ implementation.
The `*__ffi` is something `savvy` related.

**Pull request merged**